### PR TITLE
Fogl 650 startup and configuration of device server fix

### PR DESCRIPTION
**Pull request by ashwinscale**
1. change foglamp_init_data.sql - COAP configuration, task name for device is COAP also
2. change device server code to retrieve plugin to load (from configuration) based on command line argument passed in

### CODE REVIEW

File: src/python/foglamp/device/server.py [hunk 1]
```diff
@@ -69,20 +69,17 @@ class Server:
...
 
+            config = await configuration_manager.get_category_all_items(category)
+
             try:
-                plugin_module = config['plugin']
+                plugin_module = config['plugin']['value']
             except KeyError:
-                # TODO: Delete me
-                plugin_module = 'foglamp.device.coap_device'
+                sys.exit(1)
```
Comment (ashwinscale) [inline src/python/foglamp/device/server.py]: Should we throw an exception instead?

File: src/sql/foglamp_init_data.sql [hunk 2]
```diff
@@ -73,7 +73,10 @@ INSERT INTO foglamp.configuration ( key, description, value )
 INSERT INTO foglamp.configuration ( key, description, value )
      VALUES ( 'SYPURGE', 'System Purge', to_jsonb( '{ "retention" : 259200, "last purge" : "' || now() || '" }' ) );
 
-
+-- COAP:  CoAP device server
+--        plugin: python module to load dynamically
+INSERT INTO foglamp.configuration ( key, description, value )
+     VALUES ( 'COAP', 'CoAP Plugin Configuration', ' { "plugin" : { "type" : "string", "value" : "foglamp.device.coap_device", "default" : "foglamp.device.coap_device", "description" : "Python module name of the plugin to load" } } ');
```
Comment (ashwinscale) [inline src/sql/foglamp_init_data.sql]: Note than this is the fully specified module to import.

File: src/python/foglamp/device/server.py (hunk 1 above)
Comment (MarkRiddoch) [reply src/python/foglamp/device/server.py]: Let's at the very least log something. The worst thing is a service that mysteriously exists when this is at a customer site - the support guy will hate you for this :-)

File: src/sql/foglamp_init_data.sql (hunk 2 above)
Comment (MarkRiddoch) [reply src/sql/foglamp_init_data.sql]: I don't get why we have the fog lamp.device.coap_device in the config. T~his is meant to be dealt with programmatically when we do the import. This is a python and code organisation issue, not something to. burden the end user with, they just want to know they are using the coap plugin, where the python code lives is our problem not theirs.

Comment (ashwinscale) [reply src/sql/foglamp_init_data.sql]: fixed

**Pull request merged**